addons/sale_pdf_quote_builder/models/sale_order.py

```diff
@@ -48,6 +48,29 @@ def _compute_is_pdf_quote_builder_available(self):
                 or order.order_line.available_product_document_ids
             )
 
+    # === ONCHANGE METHODS === #
+
+    @api.onchange('sale_order_template_id')
+    def _onchange_sale_order_template_id(self):
+        super()._onchange_sale_order_template_id()
+        for order in self:
+            # available_quotation_document_ids = self.env['quotation.document'].search(
+            #     self.env['quotation.document']._check_company_domain(order.company_id),
+            #     order='sequence',
+            # ).filtered(lambda doc:
+            #     order.sale_order_template_id in doc.quotation_template_ids
+            #     or not doc.quotation_template_ids
+            # )
+            available_quotation_document_ids = self.env['quotation.document'].search(
+                self.env['quotation.document']._check_company_domain(order.company_id) + [
+                    ('quotation_template_ids', 'in', [False, order.sale_order_template_id.id])
+                ],
+                order='sequence',
+            )
+
+            # Remove documents which are no longer available.
+            self.quotation_document_ids &= available_quotation_document_ids
```

Comment:
```suggestion
            self.quotation_document_ids &= available_quotation_document
```
`*_ids` variable names should be reserved for sequences of actual ids, for recordset, it should be left out.